pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -165,31 +167,58 @@ func (r *Range) CIDR() net.IPNet {
 // or has already been reserved.  ErrFull will be returned if there
 // are no addresses left.
 func (r *Range) Allocate(ip net.IP) error {
+	label := r.CIDR()
```

Comment:
why not doing a simple:

```
// elevate allocated declaration 
var allocated bool
var err error
defer reportMetric(r.cidr, err)

// create a new reportMetric func and use it from both Allocate(), and AllocateNext() 
```

Reply:
I've started with that, but then I've added errors, and reportMetric is not covering all metrics (errors), so I went back to add each metrics manually